src/installation/guides/chroot.md

```diff
@@ -0,0 +1,299 @@
+
+For example, if `/dev/sda2` is to be mounted as `/` and `dev/sda1` is the EFI
+System Partition
+
+```
+# mount /dev/sda2 /mnt/
+# mkdir -p /mnt/boot/efi/
+# mount /dev/sda1 /mnt/boot/efi/
+```
+
+Initialize swap space, if desired, using
+[mkswap(8)](https://man.voidlinux.org/mkswap.8).
+
+## Base Installation
+
+This step is separated into the XBPS method and the ROOTFS method. Follow only
+one of the two following subsections.
+
+### The XBPS Method
+
+[Select an XBPS mirror.](../../xbps/repositories/mirrors/index.md) For
+simplicity, save this URL to a shell variable for later use. Append `/current`
+to this repository url to install a glibc distribution, append `/current/musl`
+for a musl distribution.
+
+> Note: musl binaries are not available for i686.
+
+For example, this is the url for glibc from the Tier 1 server in Finland,
+
+```
+# REPO=https://alpha.de.repo.voidlinux.org/current
+```
+
+xbps also needs to know what architecture is being installed, available options
+are `x86_64`, `x86_64-musl` and `i686` for PC architecture computers.
+
+For example,
+
+```
+# ARCH=x86_64
+```
+
+This architecture must be compatible with your current operating system, but
+does not need to be the same. If your host is running an x86_64 operating
+system, any of the three architectures can be installed (whether the host is
+musl or glibc), but an i686 host can only install i686 distributions.
+
+Use [xbps-install](https://man.voidlinux.org/xbps-install.1) to boot strap the
+distribution by installing the `base-system` metapackage.
+
+```
+# XBPS_ARCH=$ARCH xbps-install -S -r /mnt -R "$REPO" base-system
+```
+
+### The ROOTFS Method
+
+[Download a ROOTFS tarball](https://alpha.de.repo.voidlinux.org/live/current/)
+matching your architecture.
+
+Unpack the tarball into the newly configured filesystems.
+
+```
+# tar xvf void-[...]-ROOTFS.tar.xz -C /mnt
+```
+
+## Configuration
+
+### Entering the Chroot
+
+Mount the pseudo-filesystems needed for a chroot,
+
+```
+# mount -t proc /proc /mnt/proc
+# mount -t sysfs /sys /mnt/sys
+# mount -o bind /dev /mnt/dev
+# mount -t devpts pts /mnt/dev/pts
+```
+
+Copy the DNS configuration into the new root so that xbps can still download new
+packages inside the chroot,
+
+```
+# cp /etc/resolv.conf /mnt/etc/
+```
+
+Chroot into the new installation
+
+```
+# PS1='(chroot) # ' chroot /mnt/ /bin/bash
+```
+
+### Update the Distribution (ROOTFS method only)
+
+> Note: This section should be skipped if following the XBPS method. Your
+> distribution is already up to date.
+
+ROOTFS images generally contain out of date software, due to being a snapshot of
+the time when they were built, and do not come with a complete `base-system`.
+Update the distribution and install `base-system`.
+
+```
+# xbps-install -Su
+# xbps-install base-system
+```
+
+> Note: the first step will likely only update xbps. Running `xbps-install -u` a
+> second time will update all other packages. However, these packages will all
```

Comment:
Recent versions of `xbps` will not auto-upgrade itself, but will instead complain to the user that `xbps` is out of date and needs upgrading. Either change the `xbps-install -Su` to `xbps-install -Su xbps && xbps-install -u` (there is no harm attempting to update `xbps` if it is already up to date) and remove the note, or change the note to
```
Note: the first step may print an error message that `xbps` is out of date and must be upgraded separately. If this message appears, first run `xbps-install -Su xbps`, then run `xbps-install -u` to finish the update.
```

Reply:
good to know